ui/app/src/utils/itemActions.ts

```diff
@@ -819,6 +819,44 @@ export const itemActionDispatcher = ({
       );
       break;
     }
+    case 'deleteController': {
+      const id = uuid();
+      dispatch(
+        pushDialog({
+          minimized: true,
+          id,
+          status: 'indeterminate',
+          title: 'Processing', // TODO: i18n?
```

Comment:
Move all this stuff to an epic. Here, we'd only dispatch deleteControllerAction({ item }). Epics have access to intl. This also solves the problem with templates because epics also have full access to state, where they can read content types.